sources/logdog/src/create_tarball.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+use crate::error;
+use crate::error::Result;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use snafu::{ensure, ResultExt};
+use tar;
+
+/// Provides a function for compressing a directory's contents into a tarball.
+
+/// Creates a tarball with all the contents of directory `dir`.
+pub(crate) fn create_tarball<P1, P2>(dir: P1, outfile: P2) -> Result<()>
+where
+    P1: AsRef<Path>,
+    P2: AsRef<Path>,
+{
+    // present an error if the output directory does not exist.
+    if let Some(outdir) = outfile.as_ref().parent() {
+        ensure!(
+            Path::new(&outdir).is_dir(),
+            error::NoOutputDirectory { path: &outdir }
+        )
+    }
+
+    // present an error if we would overwrite an existing file.
+    ensure!(
+        !Path::new(outfile.as_ref()).exists(),
+        error::OutputFileExists {
+            path: PathBuf::from(outfile.as_ref()),
+        }
+    );
+
```

Comment:
```suggestion
```

I expect people to run logdog multiple times to get updated logs. Forcing them to rename the file and change their ssh workflow seems unnecessary; I think we should overwrite.